include/benchmark/benchmark_api.h

```diff
@@ -429,6 +553,33 @@ class State {
   BENCHMARK_ALWAYS_INLINE
   size_t iterations() const { return total_iterations_; }
 
+
+  // Add a user counter. Return the counter handle.
+  BENCHMARK_ALWAYS_INLINE
+  size_t SetCounter(const char* n) { return counters_.Set(n); }
```

Comment:
Fair point. As you were writing this, I was finishing up a commit changing the interface to something map-like. Maybe this way the users will be more alerted to potential costs. Please inspect that and if you still consider the access by name confusing I'll remove it.

FWIW, I still think that offering access by name is practical for the user, provided he is alert to the cost. For instance, accessing by name inside the KeepRunning loop is not a problem if the code being benchmarked is heavy in comparison and will result in few iterations.